core/src/main/scala/com/carto/analyticstoolbox/spark/spatial/OptimizeSpatial.scala

```diff
@@ -42,25 +45,30 @@ object OptimizeSpatial extends Serializable {
     // overwrite the output table
     try ssc.sql(s"DROP TABLE $outputTable;") catch { case _: AnalysisException => // e.printStackTrace() }*/
 
+    // Decide, based on column type of geometry, which parsing function to use
+    val df = ssc.table(sourceTable)
+    val parseGeom =
+      if (df.schema(geomColumn).dataType == BinaryType) ""
+      else "ST_geomFromWKT"
+
     // view creation
     // SQL definition is easier and more readable
     ssc.sql(
       s"""
          |CREATE TEMPORARY VIEW ${sourceTable}_idx_view AS(
          |  WITH orig_q AS (
          |    SELECT
-         |      * EXCEPT(geom),
-         |      geom AS wkt,
-         |      ST_geomFromWKT(geom) AS geom
+         |      * EXCEPT($geomColumn),
+         |      $parseGeom($geomColumn) AS geom
          |      FROM $sourceTable
          |    )
          |    SELECT
          |      *,
-         |      st_z2LatLon(geom) AS z2,
-         |      st_extentFromGeom(geom) AS bbox,
-         |      st_partitionCentroid(geom, $zoom) AS partitioning
+         |      st_z2LatLon(geom) AS __carto_z2,
+         |      st_extentFromGeom(geom) AS __carto_index,
+         |      st_partitionCentroid(geom, $zoom) AS __carto_partitioning
          |      FROM orig_q
-         |      DISTRIBUTE BY partitioning SORT BY z2.min, z2.max
+         |      DISTRIBUTE BY __carto_partitioning SORT BY __carto_z2.min, __carto_z2.max
```

Comment:
These changes are not affected in the `blockSizeCompute` function: https://github.com/CartoDB/analytics-toolbox-databricks/blob/7976ef8d6f276e55f21d45b39371994b6d187b1e/core/src/main/scala/com/carto/analyticstoolbox/spark/spatial/OptimizeSpatial.scala#L120-L125